### PR TITLE
[SPARK-38700][SQL][3.3] Use error classes in the execution errors of save mode

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
Migrate the following errors in QueryExecutionErrors:
 
* unsupportedSaveModeError -> UNSUPPORTED_SAVE_MODE

### Why are the changes needed?
Porting execution errors of unsupported saveMode to new error framework.
 
### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add new UT.

### CODE REVIEW

**Comment by panbingkun** [comment]
master branch pr: https://github.com/apache/spark/pull/36350
@MaxGekk ping

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, please.

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

sorry, I've been busy for the last two weeks, i will do it the weekend.

**Comment by panbingkun** [comment]
ping @MaxGekk, I have finished it

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala [hunk 1]
```diff
@@ -61,13 +77,25 @@ private[spark] object SparkThrowableHelper {
...
+      val subErrorClass = messageParameters.head
+      val errorSubInfo = subClass.getOrElse(subErrorClass,
+        throw new IllegalArgumentException(s"Cannot find sub error class '$subErrorClass'"))
+      (errorClass + "." + subErrorClass, messageParameters.tail,
+        errorInfo.messageFormat + " " + errorSubInfo.messageFormat)
+    }
+    val displayMessage = String.format(
+      displayFormat.replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
+      displayMessageParameters : _*)
     val displayQueryContext = if (queryContext.isEmpty) {
```
Comment (MaxGekk) [inline core/src/main/scala/org/apache/spark/ErrorInfo.scala:93]: Could you not include the changes into the PR. If you you changes strongly require them, let's don't backport it.

Comment (panbingkun) [reply core/src/main/scala/org/apache/spark/ErrorInfo.scala:93]: ok, i will close it and open new pr for it!
new pr: https://github.com/apache/spark/pull/36852

**Pull request closed without merging**